.github/workflows/build.yml

```
@@ -202,8 +202,8 @@ jobs:
           exit 1
         fi
 
-        # Check that commit contains ':'
-        if commit_titles |  grep -E -v '^([.a-z]+, )*[.a-z]+: [a-z]'; then
+        # Check that commit contains ':' (and no '!')
+        if commit_titles | grep -E -v '^([.a-z\-]+, )*[.a-z\-]+: [a-z\-]'; then
```

- The title should be in lowercase
- We can separate multiple components using `:` in case of hierarchical relationships

```suggestion
        if commit_titles | grep -E -v '^([.a-z\-]+[,:] )*[.a-z\-]+: [a-z][a-z0-9 _.,\-]*$'; then
```

---

I changed regex in https://github.com/osrd-project/osrd/pull/6528/commits/eff1be7da3565197350edc95a7860f86c58cc38e (a little bit more than your suggestion, as discussed)